runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java

```diff
@@ -448,45 +457,93 @@ public void onUserJoin(final UserJoin event)
 	@Subscribe
 	public void onUserSync(final UserSync event)
 	{
-		checkStateChanged(true);
+		clientThread.invokeLater(() -> checkStateChanged(true));
 		lastLocation = null;
 	}
 
 	private void checkStateChanged(boolean forceSend)
 	{
-		final int currentHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
-		final int currentPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
-		final int realHealth = client.getRealSkillLevel(Skill.HITPOINTS);
-		final int realPrayer = client.getRealSkillLevel(Skill.PRAYER);
+		if (lastStatus == null)
+		{
+			forceSend = true;
+		}
+
 		final PartyMember localMember = party.getLocalMember();
+		if (localMember == null)
+		{
+			return;
+		}
+
+		final int healthCurrent = client.getBoostedSkillLevel(Skill.HITPOINTS);
+		final int prayerCurrent = client.getBoostedSkillLevel(Skill.PRAYER);
+		final int healthMax = client.getRealSkillLevel(Skill.HITPOINTS);
+		final int prayerMax = client.getRealSkillLevel(Skill.PRAYER);
+		final int runEnergy = (int) Math.ceil(client.getEnergy() / 10.0) * 10; // flatten to reduce network load
+		final int specEnergy = client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT) / 10;
+		final boolean vengActive = client.getVarbitValue(Varbits.VENGEANCE_ACTIVE) == 1;
 
 		final Player localPlayer = client.getLocalPlayer();
 		final String characterName = Strings.nullToEmpty(localPlayer != null && client.getGameState().getState() >= GameState.LOADING.getState() ? localPlayer.getName() : null);
 
-		if (localMember != null)
+		boolean shouldSend = false;
+		final StatusUpdate update = new StatusUpdate();
+		if (forceSend || !characterName.equals(lastStatus.getN()))
```

Comment:
this NPEs

Reply:
`characterName` is guarded by the above `Strings.nullToEmpty`, and `lastStatus` is guarded by `forceSend`